test/test.go

```diff
@@ -78,6 +79,14 @@ func GetFileContent(t *testing.T, path string, def string) (exists bool, content
 func CompileBinary(t *testing.T, entrypointDir string, tempDir string) string {
 	t.Helper()
 
+	// Prevent parallel compilation, it doesn't work with the Go cache is empty
+	l := fslock.New(filepath.Join(os.TempDir(), "split-processor-compilation.lock"))
+	require.NoError(t, l.Lock())
+	defer func() {
+		require.NoError(t, l.Unlock())
+	}()
+
+	// Run build command
```

Comment:
V CI je na zaciatku Go cache prazdna.

Priblizne naraz sa zacne kompilovat processor aj CLI pre E2E testy, a spadne to, kedze oba processy naraz stahuju `go modules`.

Ak su modules uz stiahnute, tak to problem nerobi. Pridal som FS lock.